File: library/src/main/java/com/getstream/sdk/chat/model/ModelType.java
package com.getstream.sdk.chat.model;

import android.text.TextUtils;

public class ModelType {
    // Channel Type
    public static final String channel_unknown = "unknown";
    public static final String channel_livestream = "livestream";
    public static final String channel_messaging = "messaging";
    public static final String channel_team = "team";
    public static final String channel_gaming = "gaming";
    public static final String channel_commerce = "commerce";

    // Message Type
    public static final String message_regular = "regular";
    public static final String message_ephemeral = "ephemeral";
    public static final String message_error = "error";
    public static final String message_failed = "failed";
    public static final String message_reply = "reply";
    public static final String message_system = "system";

    // Attachment Type
    public static final String attach_image = "image";
    public static final String attach_imgur = "imgur";
    public static final String attach_giphy = "giphy";
    public static final String attach_video = "video";
    public static final String attach_product = "product";
    public static final String attach_file = "file";
    public static final String attach_link = "link";
    public static final String attach_unknown = "unknown";

    // File Mime Type
    public static final String attach_mime_tar = "application/tar";
    public static final String attach_mime_zip = "application/zip";

    public static final String attach_mime_txt = "text/plain";
    public static final String attach_mime_doc = "application/msword";
    public static final String attach_mime_docx = "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
    //    public static final String file_xls = "application/vnd.ms-excel";
    public static final String attach_mime_xlsx = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
    public static final String attach_mime_csv = "application/vnd.ms-excel";
    public static final String attach_mime_ppt = "application/vnd.ms-powerpoint";
    public static final String attach_mime_pdf = "application/pdf";

    public static final String attach_mime_mov = "video/mov";
    public static final String attach_mime_mp4 = "video/mp4";
    public static final String attach_mime_mp3 = "audio/mp3";
    public static final String attach_mime_m4a = "audio/m4a";
    // Action Type
    public static final String action_send = "send";
    public static final String action_shuffle = "shuffle";
    public static final String action_cancel = "cancel";

    public static String getAssetUrl(Attachment attachment){
        switch (attachment.getType()) {
            case ModelType.attach_image:
                return attachment.getImageURL();
            case ModelType.attach_giphy:
            case ModelType.attach_video:
                return attachment.getThumbURL();
            default:
                return TextUtils.isEmpty(attachment.getImageURL()) ? attachment.getImage() : attachment.getImageURL();
        }
    }
}

File: library/src/main/java/io/getstream/chat/android/client/models/ModelType.kt
package io.getstream.chat.android.client.models

import android.text.TextUtils


object ModelType {
    // Channel Type
    const val channel_unknown = "unknown"
    const val channel_livestream = "livestream"
    const val channel_messaging = "messaging"
    const val channel_team = "team"
    const val channel_gaming = "gaming"
    const val channel_commerce = "commerce"
    // Message Type
    const val message_regular = "regular"
    const val message_ephemeral = "ephemeral"
    const val message_error = "error"
    const val message_failed = "failed"
    const val message_reply = "reply"
    const val message_system = "system"
    // Attachment Type
    const val attach_image = "image"
    const val attach_imgur = "imgur"
    const val attach_giphy = "giphy"
    const val attach_video = "video"
    const val attach_product = "product"
    const val attach_file = "file"
    const val attach_link = "link"
    const val attach_unknown = "unknown"
    // File Mime Type
    const val attach_mime_tar = "application/tar"
    const val attach_mime_zip = "application/zip"
    const val attach_mime_txt = "text/plain"
    const val attach_mime_doc = "application/msword"
    const val attach_mime_docx =
        "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
    //    public static final String file_xls = "application/vnd.ms-excel";
    const val attach_mime_xlsx =
        "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
    const val attach_mime_csv = "application/vnd.ms-excel"
    const val attach_mime_ppt = "application/vnd.ms-powerpoint"
    const val attach_mime_pdf = "application/pdf"
    const val attach_mime_mov = "video/mov"
    const val attach_mime_mp4 = "video/mp4"
    const val attach_mime_mp3 = "audio/mp3"
    // Action Type
    const val action_send = "send"
    const val action_shuffle = "shuffle"
    const val action_cancel = "cancel"
    fun getAssetUrl(attachment: Attachment): String? {
        return when (attachment.type) {
            attach_image -> attachment.imageURL
            attach_giphy, attach_video -> attachment.thumbURL
            else -> if (TextUtils.isEmpty(attachment.imageURL)) attachment.image else attachment.imageURL
        }
    }
}
